src/DatePicker.js

```diff
@@ -1,41 +1,29 @@
-import React from 'react';
+import React, { useMemo, useEffect, useRef } from 'react';
 import PropTypes from 'prop-types';
 import TextInput from './TextInput';
 import idgen from './idgen';
 
-class DatePicker extends React.Component {
-  constructor(props) {
-    super(props);
+const DatePicker = props => {
+  const id = useMemo(() => props.id || `datepicker${idgen()}`, [props.id]);
```

Comment:
You know what i've seen a lot lately? making props.id a default value of `datepicker${idgen()`, WDYT?

Reply:
Do you mean something like that:
"const DatePicker = ({ id:  `datepicker${idgen()}`, ...props}) => {" ?

Reply:
Do you mean something like that:
const DatePicker = ({ id = `datepicker${idgen()}`, ...props }) => {
?